src/camera.rs

```diff
@@ -432,3 +266,510 @@ impl Default for capi::CameraProperties_camera_properties_shape_s {
+
+impl Default for TriggerInputOutputCapabilities {
+    fn default() -> Self {
+        Self {
+            input: Default::default(),
+            output: Default::default(),
+        }
+    }
+}
+
+impl TryFrom<capi::CameraPropertyMetadata_CameraPropertiesTriggerMetadata_camera_properties_metadata_trigger_capabilities_s> for TriggerInputOutputCapabilities {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: capi::CameraPropertyMetadata_CameraPropertiesTriggerMetadata_camera_properties_metadata_trigger_capabilities_s,
+    ) -> Result<Self, Self::Error> {
+        Ok(Python::with_gil(|_| -> PyResult<_> {
+            Ok(Self {
+                input: value.input,
+                output: value.output,
+            })
+        })?)
+    }
+}
+
+/// CameraCapabilities::TriggerCapabilities
+#[pyclass]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TriggerCapabilities {
+    #[pyo3(get, set)]
+    acquisition_start: Py<TriggerInputOutputCapabilities>,
+
+    #[pyo3(get, set)]
+    exposure: Py<TriggerInputOutputCapabilities>,
+
+    #[pyo3(get, set)]
+    frame_start: Py<TriggerInputOutputCapabilities>,
+}
+
+impl_plain_old_dict!(TriggerCapabilities);
+
+impl Default for TriggerCapabilities {
+    fn default() -> Self {
+        let (acquisition_start, exposure, frame_start) = Python::with_gil(|py| {
+            (
+                Py::new(py, TriggerInputOutputCapabilities::default()).unwrap(),
+                Py::new(py, TriggerInputOutputCapabilities::default()).unwrap(),
+                Py::new(py, TriggerInputOutputCapabilities::default()).unwrap(),
+            )
+        });
+        Self {
+            acquisition_start,
+            exposure,
+            frame_start,
+        }
+    }
+}
+
+impl TryFrom<capi::CameraPropertyMetadata_CameraPropertiesTriggerMetadata> for TriggerCapabilities {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: capi::CameraPropertyMetadata_CameraPropertiesTriggerMetadata,
+    ) -> Result<Self, Self::Error> {
+        let (acquisition_start, exposure, frame_start) = Python::with_gil(|py| -> PyResult<_> {
+            let acquisition_start: TriggerInputOutputCapabilities =
+                value.acquisition_start.try_into()?;
+            let exposure: TriggerInputOutputCapabilities = value.exposure.try_into()?;
+            let frame_start: TriggerInputOutputCapabilities = value.frame_start.try_into()?;
+            Ok((
+                Py::new(py, acquisition_start)?,
+                Py::new(py, exposure)?,
+                Py::new(py, frame_start)?,
+            ))
+        })?;
+        Ok(Python::with_gil(|_| -> PyResult<_> {
+            Ok(Self {
+                acquisition_start,
+                exposure,
+                frame_start,
+            })
+        })?)
+    }
+}
+
+/// CameraCapabilities
+#[pyclass]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CameraCapabilities {
+    #[pyo3(get, set)]
```

Comment:
Should most (maybe all?) of the Python attributes/properties in this PR be read-only (i.e. `pyo3(get)`)?

Or do we need the `set` internally?

Reply:
We don't actually need `set` internally. Will fix.